src/main/java/com/skeleton/feed/service/StatisticsService.java

```diff
@@ -0,0 +1,59 @@
+package com.skeleton.feed.service;
+
+import com.skeleton.common.exception.CustomException;
+import com.skeleton.common.exception.ErrorCode;
+import com.skeleton.common.util.DateStatisticsCalculator;
+import com.skeleton.common.util.HourStatisticsCalculator;
+import com.skeleton.common.util.StatisticsCalculator;
+import com.skeleton.feed.dto.StatisticsRequest;
+import com.skeleton.feed.dto.StatisticsResponse;
+import com.skeleton.feed.entity.Statistics;
+import com.skeleton.feed.repository.StatisticsRepository;
+import jakarta.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class StatisticsService {
+    private final StatisticsRepository repository;
+    private final List<StatisticsCalculator> calculatorList;
+    private final Map<String, StatisticsCalculator> calculators = new HashMap<>();
+
+    @PostConstruct
+    public void init() {
+        calculators.putAll(calculatorList.stream()
+                .collect(Collectors.toMap(this::getCalculatorType, Function.identity())));
+    }
+
+    private String getCalculatorType(StatisticsCalculator calculator) {
+        if (calculator instanceof DateStatisticsCalculator) {
+            return "date";
+        }
+        if (calculator instanceof HourStatisticsCalculator) {
+            return "hour";
+        }
+        throw new CustomException(ErrorCode.TYPE_NOT_FOUND);
+    }
+
+    public StatisticsResponse getStatistics(StatisticsRequest request, Authentication authentication) {
+        String userId = ((UserDetails) authentication.getPrincipal()).getUsername();
```

Comment:
```suggestion
        String userEmail= ((UserDetails) authentication.getPrincipal()).getUsername();
```

userEmail이 좀 더 정확한 명칭인 것 같습니다. 

애초에 User 엔티티 의 id는 `Long` 타입이라서 타입이 맞지 않고, 

또 `getStatistics` 메서드 내 `filter`에서는 String userId를 getEmail과 비교합니다. 이걸 본 후에야 해당 String이 User엔티티의 Id(PK)가 아닌 이메일 이라는 걸 알 수 있었네요. 😅 

- filter(stat -> `userId` .equals(stat.getUserId().`getEmail()`)

오해를 줄이기 위해 다음부터는 명명에 좀 더 신경써주시면 좋겠어요👍